### DISCUSSION TITLE
[lodash] `!isEmpty()` undesired type widening

### DISCUSSION THREAD

**Discussion by judfs** [DefinitelyTyped/DefinitelyTyped#62379] [Issues with a @types package]
Changes in `4.14.183` to `isEmpty` caused my build to break. Changes: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/61620 @forivall 

```typescript
import * as _ from "lodash";

let bag = {};
// bag["foo"] = 1;
let bag2;

if (!_.isEmpty(bag)) {
    bag2 = { ...bag };
}  
```
```
spread.ts:8:14 - error TS2698: Spread types may only be created from object types.

8     bag2 = { ...bag };
```

In `4.14.182` this example compiles without error. The example can be fixed for 183 by saying `let bag: Record<string, any> = {};` instead. I'm using `"strict": false` so otherwise I wouldn't bother specifying the type.

Edit: related: https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/62155
I only searched issues, didn't know to search for discussions.

[a comment by an automated account is not shown]

**Comment by forivall** [comment]
Technically, your example doesn't make sense, since the body of the `if` statement is unreachable code, and TypeScript is correctly failing; although, in an unclear manner.

Nevertheless, https://github.com/DefinitelyTyped/DefinitelyTyped/pull/62409 solves this.

**Reply by aj-r** [reply]
I agree, @judfs can you give a more complete example? Checking if an empty object is empty seems pointless so I expect there is something else going on.

**Reply by judfs** [reply]
The spread error persists if you uncomment the foo index assignment (and have strict off to allow that). Could just as well hide that assignment behind `if(Math.random() <0.5)`. Just going for the minimal example. Also using variable instead of literal indexes in my actual usage.

Typescript is otherwise happy to allow the pointless operation `let a = {}; let b = {..a};`.

Because of the widening, trying to assign to `bag` in the `isEmpty` block is also an error:

```
spread.ts:10:1 - error TS2322: Type 'number' is not assignable to type 'never'.

10 bag["bar"] = 1;
```

Anyway if fixes inspired by the other issue fix this too then awesome!

**Reply by aj-r** [reply]
In that case you really should specify the type (`bag: Record<string, any>`). Normally `bag["bar"] = 1` would be an error without specifying the type - I'm guessing it only works because you have `"strict": false`. So it sounds like this is WAI.
